src/engine/source/builder/builders/opBuilderKVDB.cpp

```diff
@@ -349,88 +364,97 @@ base::Expression KVDBSet(const std::string& targetField,
-                                                  event->setBool(false, targetField);
-
-                                                  // Get DB name
-                                                  std::string resolvedDBName;
-                                                  if (Parameter::Type::REFERENCE == dbName.m_type)
-                                                  {
-                                                      const auto retval = event->getString(dbName.m_value);
-                                                      if (retval)
-                                                      {
-                                                          resolvedDBName = retval.value();
-                                                      }
-                                                      else
-                                                      {
-                                                          return base::result::makeFailure(event, failureTrace1);
-                                                      }
-                                                  }
-                                                  else
-                                                  {
-                                                      resolvedDBName = dbName.m_value;
-                                                  }
-
-                                                  const auto deleteResult = kvdbManager->deleteDB(resolvedDBName);
-                                                  if (deleteResult)
-                                                  {
-                                                      return base::result::makeFailure(
-                                                          event,
-                                                          failureTrace
-                                                              + fmt::format("Database '{}' could not be deleted: {}",
-                                                                            resolvedDBName,
-                                                                            deleteResult.value().message));
-                                                  }
-
-                                                  event->setBool(true, targetField);
-
-                                                  return base::result::makeSuccess(event, successTrace);
-                                              });
+    return base::Term<base::EngineOp>::create(
+        name,
+        [=,
+         targetField = std::move(targetField),
+         kvdbHandler =
+             std::move(std::get<std::shared_ptr<kvdbManager::IKVDBHandler>>(resultHandler))](base::Event event)
+        {
+            event->setBool(false, targetField);
+
+            std::string resolvedKey;
+            if (Parameter::Type::REFERENCE == key.m_type)
+            {
+                const auto value = event->getString(key.m_value);
+                if (value)
```

Comment:
It is the same, since std::optional provides a [bool operator](https://en.cppreference.com/w/cpp/utility/optional/operator_bool) to achieve this. But to keep clarity and consistency with the rest of the Engine's code, we will change this to has_value(). 
Thank you!